hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -551,13 +557,14 @@ public static long dateToLong(final Date date) {
   /**
    * The standard AWS provider list for AWS connections.
    */
+  @SuppressWarnings("deprecation")
   public static final List<Class<?>>
       STANDARD_AWS_PROVIDERS = Collections.unmodifiableList(
       Arrays.asList(
           TemporaryAWSCredentialsProvider.class,
           SimpleAWSCredentialsProvider.class,
           EnvironmentVariableCredentialsProvider.class,
-          IAMInstanceCredentialsProvider.class));
```

Comment:
using fully qualified class names to avoid deprecation warnings that happen on importing a deprecated class. not sure if there is a better way to suppress import warnings?

Reply:
we just ignore them if there's no way to avoid